storage/redis_cluster.go

```diff
@@ -115,35 +52,38 @@ func getRedisAddrs(config config.StorageOptionsConf) (addrs []string) {
 	return addrs
 }
 
-func clusterConnectionIsOpen(cluster *RedisCluster) bool {
-
-	c := cluster.RedisController.singleton(cluster.IsCache, cluster.IsAnalytics)
-	testKey := "redis-test-" + uuid.NewV4().String()
-	if err := c.Set(cluster.RedisController.ctx, testKey, "test", time.Second).Err(); err != nil {
-		return false
-	}
-	if _, err := c.Get(cluster.RedisController.ctx, testKey).Result(); err != nil {
-		return false
-	}
-	return true
-}
-
 // Connect will establish a connection this is always true because we are
 // dynamically using redis
 func (r *RedisCluster) Connect() bool {
 	return true
 }
 
-func (r *RedisCluster) singleton() redis.UniversalClient {
+func (r *RedisCluster) singleton() RedisDriver {
 	return r.RedisController.singleton(r.IsCache, r.IsAnalytics)
 }
 
-func (r *RedisCluster) hashKey(in string) string {
+func (r *RedisCluster) check(err error, message string, opt ...interface{}) {
+	isDebug := os.Getenv("TYK_LOGLEVEL") == "debug"
+
+	// Turn on extra debug output with env
+	if err != nil || isDebug {
+		label := "RedisCluster." + message + ":"
+		message := make([]interface{}, 0, 1+len(opt)*2)
+		message = append(message, label)
+		for _, o := range opt {
+			message = append(message, " ")
+			message = append(message, o)
+		}
+		log.WithError(err).Debug(message...)
```

Comment:
@titpetric If the logger is set on Error level, it won't log anything here. Or Am I missing something?

Reply:
You are probably not wrong. The design issue with storage is that it doesn't return errors and cannot handle them. Logging of any sort in a driver level interface is bad, as this could reasonably flood the error output with noisy unactionable data. Assuming the frequency of the output is high, the user should explicitly opt-into a debug error level. I concede this is a discussion point, feel free to weigh in with a sensible suggestion 😂